### PR TITLE
[test] Change yt-dlp failures to warnings, validate script/server output directly 

### PR DESCRIPTION
**Pull request by Brainicism**
- Test script output is correct (non-null POT and visitor ID)
- Test server response is correct (non-null POT and visitor ID)
- Change yt-dlp failures to warnings (failures are expected due to bot check)

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -54,6 +54,18 @@ jobs:
         run: |
           exit_code=0
 
+          # check that the script is returning a POT
+          script_output=$(node server/build/generate_once.js)
+
+          po_token=$(echo "$script_output" | jq -r '.po_token')
+          visit_identifier=$(echo "$script_output" | jq -r '.visit_identifier')
+
+          if [[ "$po_token" != "null" && "$visit_identifier" == "null" ]]; then
```
Comment (grqx) [inline .github/workflows/test.yml:63]:
```suggestion
          if [[ "$po_token" == "null" || "$visit_identifier" == "null" ]]; then
```

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -123,6 +134,17 @@ jobs:
           echo "$PING_RESP" | jq -C
           echo "::endgroup::"
 
+          # check that the server is returning a POT
+          pot_response=$(curl -s -X POST "http://127.0.0.1:${{ env.PORT }}/get_pot")
+          po_token=$(echo "$pot_response" | jq -r '.po_token')
+          visit_identifier=$(echo "$pot_response" | jq -r '.visit_identifier')
+
+          if [[ "$po_token" != "null" && "$visit_identifier" == "null" ]]; then
```
Comment (grqx) [inline .github/workflows/test.yml:142]:
```suggestion
          if [[ "$po_token" == "null" || "$visit_identifier" == "null" ]]; then
```

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -123,6 +134,17 @@ jobs:
...
+              echo "Error: 'po_token' or 'visit_identifier' is null."
```
Comment (grqx) [inline .github/workflows/test.yml:143]:
```suggestion
              echo "::error title=PoToken generation failed,file=server/src/session_manager.ts::'po_token' or 'visit_identifier' is null."
```

File: .github/workflows/test.yml [hunk 4]
```diff
@@ -54,6 +54,18 @@ jobs:
...
           exit_code=0
 
+          # check that the script is returning a POT
+          script_output=$(node server/build/generate_once.js)
+
+          po_token=$(echo "$script_output" | jq -r '.po_token')
+          visit_identifier=$(echo "$script_output" | jq -r '.visit_identifier')
+
+          if [[ "$po_token" != "null" && "$visit_identifier" == "null" ]]; then
+              echo "Error: 'po_token' or 'visit_identifier' is null."
```
Comment (grqx) [inline .github/workflows/test.yml:64]:
```suggestion
              echo "::error title=PoToken generation failed,file=server/src/session_manager.ts::'po_token' or 'visit_identifier' is null."
```

File: .github/workflows/test.yml [hunk 5]
```diff
@@ -123,6 +134,17 @@ jobs:
           echo "$PING_RESP" | jq -C
           echo "::endgroup::"
 
+          # check that the server is returning a POT
+          pot_response=$(curl -s -X POST "http://127.0.0.1:${{ env.PORT }}/get_pot")
```
Comment (grqz) [inline .github/workflows/test.yml:138]: lets log this
```suggestion
          pot_response=$(curl -s -X POST "http://127.0.0.1:${{ env.PORT }}/get_pot")
          echo "::group::Get pot response from the HTTP server"
          echo "$pot_response" | jq -C
          echo "::endgroup::"
```

File: .github/workflows/test.yml [hunk 6]
```diff
@@ -54,6 +54,18 @@ jobs:
         run: |
           exit_code=0
 
+          # check that the script is returning a POT
+          script_output=$(node server/build/generate_once.js)
```
Comment (grqz) [inline .github/workflows/test.yml:58]:
```suggestion
          script_output=$(node server/build/generate_once.js)
          echo "::group::POT response from the script"
          echo "$script_output" | jq -C
          echo "::endgroup::"
```

**Review by grqz — APPROVED** [review]
LGTM

**Pull request merged**